app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt

```diff
@@ -0,0 +1,126 @@
+          DataBindingUtil.inflate<WelcomeBinding>(
+            inflater,
+            R.layout.welcome,
+            parent,
+            /* attachToParent= */false
+          )
+        WelcomeViewHolder(binding)
+      }
+      VIEW_TYPE_PROMOTED_STORY -> {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding =
+          DataBindingUtil.inflate<PromotedStoryCardBinding>(
+            inflater,
+            R.layout.promoted_story_card,
+            parent,
+            /* attachToParent= */false
+          )
+        PromotedStoryViewHolder(binding)
+      }
+      VIEW_TYPE_TOPIC_LIST -> {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding =
+          DataBindingUtil.inflate<TopicSummaryViewBinding>(
+            inflater,
+            R.layout.topic_summary_view,
+            parent,
+            /* attachToParent= */false
+          )
+        TopicListViewHolder(binding)
+      }
+      else -> throw IllegalArgumentException("Invalid view type") as Throwable
+    }
+  }
+
+  override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+    when (holder.itemViewType) {
+      VIEW_TYPE_WELCOME_MESSAGE -> {
+        (holder as WelcomeViewHolder).bind(itemList[position] as UserAppHistoryViewModel)
+      }
+      VIEW_TYPE_PROMOTED_STORY -> {
+        (holder as PromotedStoryViewHolder).bind(itemList[position] as PromotedStoryViewModel)
+      }
+      VIEW_TYPE_TOPIC_LIST -> {
+        (holder as TopicListViewHolder).bind(itemList[position] as TopicSummary, topicSummaryClickListener)
+      }
+    }
+  }
+
+  override fun getItemViewType(position: Int): Int {
+    return when (itemList[position]) {
+      is UserAppHistoryViewModel -> {
+        VIEW_TYPE_WELCOME_MESSAGE
+      }
+      is PromotedStoryViewModel -> {
+        VIEW_TYPE_PROMOTED_STORY
+      }
+      is TopicSummary -> {
+        VIEW_TYPE_TOPIC_LIST
+      }
+      else -> throw IllegalArgumentException("Invalid type of data $position")
+    }
+  }
+
+  override fun getItemCount(): Int {
+    return itemList.size
+  }
+
+  private class WelcomeViewHolder(
+    val binding: ViewDataBinding
+  ) : RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(userAppHistoryViewModel: UserAppHistoryViewModel) {
+      binding.setVariable(BR.viewModel, userAppHistoryViewModel)
+    }
+  }
+
+  private class PromotedStoryViewHolder(
+    val binding: ViewDataBinding
+  ) : RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(promotedStoryViewModel: PromotedStoryViewModel) {
+      binding.setVariable(BR.viewModel, promotedStoryViewModel)
+    }
+  }
+
+  private class TopicListViewHolder(
+    val binding: ViewDataBinding
+  ) : RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(topicSummary: TopicSummary, topicSummaryClickListener: TopicSummaryClickListener) {
+
+      val topicSummaryViewModel = TopicSummaryViewModel(topicSummary, topicSummaryClickListener)
```

Comment:
I suggest binding TopicSummaryViewModel in the item list instead of TopicSummary. That makes this cleaner and more consistent with the other view holders (which consequentially makes the migration to the bindable adapter trivial).

Reply:
Updated to bind all this and pass it to itemList similar to implementation in other Items of the list.